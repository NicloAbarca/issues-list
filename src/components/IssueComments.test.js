import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { checkSnapshot } from '../utils/testUtils';
import IssueComments from './IssueComments';

Enzyme.configure({ adapter: new Adapter() });

it('renders with no comments', () => {
  const tree = shallow(<IssueComments/>);
  checkSnapshot(tree);
});

it('renders comments', () => {
  const comment = {
    user: {
      login: "somebody",
      avatar_url: "http://foo"
    },
    body: "a comment"
  };
  const comments = [{...comment, id: 1}, {...comment, id: 2}];
  const tree = shallow(
    <IssueComments comments={comments}/>
    );
  checkSnapshot(tree);
});