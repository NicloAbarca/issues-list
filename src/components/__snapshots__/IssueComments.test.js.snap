// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders comments 1`] = `
<ul
  className="issue-detail__comments"
>
  <li
    key="1"
  >
    <IssueComment
      comment={
        Object {
          "body": "a comment",
          "id": 1,
          "user": Object {
            "avatar_url": "http://foo",
            "login": "somebody",
          },
        }
      }
    />
  </li>
  <li
    key="2"
  >
    <IssueComment
      comment={
        Object {
          "body": "a comment",
          "id": 2,
          "user": Object {
            "avatar_url": "http://foo",
            "login": "somebody",
          },
        }
      }
    />
  </li>
</ul>
`;

exports[`renders with no comments 1`] = `
<ul
  className="issue-detail__comments"
/>
`;

exports[`test renders comments 1`] = `
<ul
  className="issue-detail__comments">
  <li>
    <IssueComment
      comment={
        Object {
          "body": "a comment",
          "id": 1,
          "user": Object {
            "avatar_url": "http://foo",
            "login": "somebody",
          },
        }
      } />
  </li>
  <li>
    <IssueComment
      comment={
        Object {
          "body": "a comment",
          "id": 2,
          "user": Object {
            "avatar_url": "http://foo",
            "login": "somebody",
          },
        }
      } />
  </li>
</ul>
`;

exports[`test renders with no comments 1`] = `
<ul
  className="issue-detail__comments" />
`;
